corgidrp/spec.py

```diff
@@ -0,0 +1,648 @@
+        clock_ang (float): Clocking angle of the dispersion axis in degrees
+        ref_wavlen (float): Reference wavelength of the bandpass, in nanometers
+        pixel_pitch_um (float): EXCAM pixel pitch in microns
+
+    Returns:
+        pfit_pos_vs_wavlen (numpy.ndarray): polynomial coefficients for the
+        position vs wavelength fit
+        cov_pos_vs_wavlen (numpy.ndarray): covariance matrix of the polynomial
+        coefficients for the position vs wavelength fit
+        pfit_wavlen_vs_pos (numpy.ndarray): polynomial coefficients for the
+        wavelength vs position fit
+        cov_wavlen_vs_pos (numpy.ndarray): covariance matrix of the polynomial
+        coefficients for the wavelength vs position fit
+    """
+    pixel_pitch_mm = pixel_pitch_um * 1E-3
+
+    # Rotate the centroid coordinates so the dispersion axis is horizontal
+    # to define a rotation pivot point, select the filter closest to the nominal 
+    # zero deviation wavelength
+    refidx = np.argmin(np.abs(wavlens - ref_wavlen))
+    (x_rot, y_rot) = rotate_points((xpts, ypts), -np.deg2rad(clock_ang), 
+                                    pivot_point = (xpts[refidx], ypts[refidx]))
+    
+    # Fit an intermediate polynomial to wavelength versus position
+    delta_x = (x_rot - x_rot[refidx]) * pixel_pitch_mm
+    pos_err = cent_errs * pixel_pitch_mm
+    weights = 1 / pos_err
+    lambda_func_x = np.poly1d(np.polyfit(x = delta_x, y = wavlens, deg = 2, w = weights))
+    # Determine the position at the reference wavelength
+    poly_roots = (np.poly1d(lambda_func_x) - ref_wavlen).roots
+    real_roots = poly_roots[np.isreal(poly_roots)]
+    root_select_ind = np.argmin(np.abs(poly_roots[np.isreal(poly_roots)]))
+    pos_ref = np.real(real_roots[root_select_ind])
+    np.testing.assert_almost_equal(lambda_func_x(pos_ref), ref_wavlen)
+    displacements_mm = delta_x - pos_ref
+
+    # Fit two polynomials:  
+    # 1. Displacement from the band center along the dispersion axis as a 
+    #    function of wavelength  
+    # 2. Wavelength as a function of displacement along the dispersion axis
+    (pfit_pos_vs_wavlen,
+     cov_pos_vs_wavlen) = np.polyfit(x = (wavlens - ref_wavlen) / ref_wavlen,
+                                      y = displacements_mm, deg = 3, w = weights, cov=True)
+    
+    (pfit_wavlen_vs_pos,
+     cov_wavlen_vs_pos) = np.polyfit(x = displacements_mm, y = wavlens, deg = 3, 
+                                      w = weights, cov=True)
+
+    return pfit_pos_vs_wavlen, cov_pos_vs_wavlen, pfit_wavlen_vs_pos, cov_wavlen_vs_pos
+
+
+def calibrate_dispersion_model(centroid_psf, band_center_file, prism = 'PRISM3', pixel_pitch_um = 13.0):
+    """ 
+    Generate a DispersionModel of the spectral dispersion profile of the CGI ZOD prism.
+
+    Args:
+       centroid_psf (SpectroscopyCentroidPsf): instance of SpectroscopyCentroidPsf calibration class
+       band_center_file (str): file name of the band centers
```

Comment:
should there be a default band_center_file that lives in corgidrp/data alongside the filter curves? so a user does not need to always pass in a file path?